src/bluefish.tsx

```diff
@@ -157,21 +176,48 @@ export function Bluefish(props: BluefishProps) {
     );
   };
 
-  return (
-    <>
+  const jsx = (
+    <LayoutUIDContext.Provider value={layoutUID}>
       <ErrorContext.Provider value={errorContext}>
         <ScenegraphContext.Provider value={scenegraphContext}>
           <ScopeContext.Provider value={[scope, setScope]}>
-            <Layout name={id} layout={layout} paint={paint}>
-              <ParentScopeIdContext.Provider value={() => scopeId}>
-                <ParentIDContext.Provider value={id}>
-                  {props.children}
-                </ParentIDContext.Provider>
-              </ParentScopeIdContext.Provider>
-            </Layout>
+            {untrack(() => {
+              const layoutNode = resolveScenegraphElements(
+                <Layout name={id} layout={layout} paint={paint}>
+                  <ParentScopeIdContext.Provider value={() => scopeId}>
+                    {props.children}
+                  </ParentScopeIdContext.Provider>
+                </Layout>
+              );
+
+              setFullLayoutFunction(() => {
+                return () => layoutNode[0].layout(null);
+              });
+
+              return layoutNode[0].jsx;
+            })}
```

Comment:
withBluefish intercepts the layout node, wraps it in some additional context, and passes it up the tree

Reply:
Thanks. My comment was ambiguous; I'm still wondering if this code could/should be factored out so that behaviour is consistent? Or do these two pieces of code just happen to look alike and could diverge in future changes?

Reply:
I'm not 100% sure they should/will be the same so I'll leave them distinct for now.